test/integration/ocsp_test.go

```diff
@@ -0,0 +1,192 @@
+		}
+
+		chal, ok := auth.ChallengeMap[acme.ChallengeTypeHTTP01]
+		if !ok {
+			return nil, fmt.Errorf("no HTTP challenge at %s", authUrl)
+		}
+
+		err = addHTTP01Response(chal.Token, chal.KeyAuthorization)
+		if err != nil {
+			return nil, fmt.Errorf("adding HTTP-01 response: %s", err)
+		}
+		chal, err = c.Client.UpdateChallenge(c.Account, chal)
+		if err != nil {
+			return nil, fmt.Errorf("updating challenge: %s", err)
+		}
+	}
+
+	certKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("generating certificate key: %s", err)
+	}
+
+	csrDer, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{
+		SignatureAlgorithm: x509.ECDSAWithSHA256,
+		PublicKeyAlgorithm: x509.ECDSA,
+		PublicKey:          certKey.Public(),
+		Subject:            pkix.Name{CommonName: domains[0]},
+		DNSNames:           domains,
+	}, certKey)
+	if err != nil {
+		return nil, fmt.Errorf("making csr: %s", err)
+	}
+	csr, err := x509.ParseCertificateRequest(csrDer)
+	if err != nil {
+		return nil, fmt.Errorf("parsing csr: %s", err)
+	}
+
+	order, err = c.Client.FinalizeOrder(c.Account, order, csr)
+	if err != nil {
+		return nil, fmt.Errorf("finalizing order: %s", err)
+	}
+	certs, err := c.Client.FetchCertificates(c.Account, order.Certificate)
+	if err != nil {
+		return nil, fmt.Errorf("fetching certificates: %s", err)
+	}
+	return &issuanceResult{order, certs}, nil
+}
+
+func TestPrecertificateOCSP(t *testing.T) {
+	domain := random_domain()
+	for _, port := range []int{4500, 4501, 4510, 4511} {
+		url := fmt.Sprintf("http://boulder:%d/add-reject-host", port)
+		body := []byte(fmt.Sprintf(`{"host": "%s"}`, domain))
+		resp, err := http.Post(url, "", bytes.NewBuffer(body))
+		if err != nil {
+			t.Fatalf("adding reject host: %s", err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("adding reject host: %d", resp.StatusCode)
+		}
+		resp.Body.Close()
+	}
+
+	os.Setenv("DIRECTORY", "http://boulder:4001/directory")
+	_, err := authAndIssue([]string{domain})
+	if err != nil {
+		if strings.Contains(err.Error(), "urn:ietf:params:acme:error:serverInternal") &&
+			strings.Contains(err.Error(), "SCT embedding") {
+		} else {
+			t.Fatal(err)
+		}
+	}
+	if err == nil {
+		t.Fatal("expected error issuing for domain rejected by CT servers; got none")
+	}
+
+	resp, err := http.Get("http://boulder:4500/get-rejections")
+	if err != nil {
+		t.Fatalf("getting rejections: %s", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("getting rejections: status %d", resp.StatusCode)
+	}
+	var rejections []string
+	err = json.NewDecoder(resp.Body).Decode(&rejections)
+	if err != nil {
+		t.Fatalf("parsing rejections: %s", err)
+	}
+
+	for _, r := range rejections {
+		rejectedCertBytes, err := base64.StdEncoding.DecodeString(r)
+		if err != nil {
+			t.Fatalf("decoding rejected cert: %s", err)
+		}
+		_, err = ocsp_helper.ReqDER(rejectedCertBytes)
```

Comment:
It's really nice that you could just change this function to be exported and use it in an integration test vs having to write a second version :tada: